addRestaurant.js

```diff
@@ -33,19 +33,23 @@ $(document).ready(function() {
 });
 
 function addRestaurant() {
+  var $addRestaurantModalForm = $('#addRestaurantModalForm');
+
   var nameRestaurant = $('#name').val();
   var adressRestaurant = $('#address').val();
   var categoryIndex = $('#category').val();
   var categoryRestaurant = categories.find(function(category) {
     return category.id == categoryIndex;
   });
 
-  if (
-    categoryRestaurant != null &&
-    nameRestaurant != null &&
-    adressRestaurant != null &&
-    latlngSearchedMap.lat !== null
-  ) {
-    //addNewRestaurant(nameRestaurant,categoryRestaurant, adressRestaurant,latlngSearchedMap.lat,latlngSearchedMap.lng)
+  if ($addRestaurantModalForm.valid() && latlngSearchedMap != null) {
+    console.log(adressRestaurant);
+    addNewRestaurant(
```

Comment:
let's call this method somehow different - i needed to check the whole file if you're not calling the same method inside the method (recursion) - maybe something that would tell us that it's sending the data to api?
like `saveRestaurant` or sth